addons/account/models/account_account_tag.py

```diff
@@ -29,3 +29,11 @@ def _validate_tag_country(self):
         for record in self:
             if record.applicability == 'taxes' and not record.country_id:
                 raise ValidationError(_("A tag defined to be used on taxes must always have a country set."))
+
+    def unlink(self):
+        external_ids = self.get_external_id()
+        for record in self:
+            external_id = external_ids[record.id]
```

Comment:
Quid if the record doesn't have any external id? I'm not sure there is an entry in external_ids in that case

Reply:
Yes , But only manually created records do not have an external id . But once they are exported with external id they do get associated to an unique external id. So manually created records would not have any entry in external_ids.

Reply:
@sami-odoo 
manually created records may get external id after exporting. You should add extra check as on this example: https://github.com/odoo/odoo/commit/7082ce2460494891e8e063c11678226b164f99e7

Reply:
@yelizariev sir the fix you require has already been done when you asked earlier for it . Please check the unlink function in my solution

Reply:
@sami-odoo you confuse ondelete attribute and ondelete decorator

![image](https://github.com/odoo/odoo/assets/186131/4294380e-106b-45e7-b1a5-0a41e32647c1)